Add required and suggested PHP extensions to composer.json

JoshuaLuckers opened the pull request:
### What does it do?
This adds required and suggested PHP extensions to composer.json.

### Why is it needed?
We want to check and suggest the system requirements when installing MODX via composer.

### Related issue(s)/PR(s)
Fixes issue #14048

JoshuaLuckers committed:
When installing MODX via composer we want to check and suggest system requirements

JoshuaLuckers committed:
Install ImageMagick on Travis

opengeek reviewed (changes requested):
I don't think ext-imagick is required for MODX is it? Is this a new requirement? Because I've not installed this extension in quite some time and am now blocked from installing. :P

alroniks commented:
Jason is right. Imagick can be replaced (and it does) by php-gd. It works better but it is not installed everywhere so we can not lock MODX to use this extension. I suggest moving it to recommended section @JoshuaLuckers

JoshuaLuckers commented:
It’s listed as one of the [server requirements in the docs](
https://docs.modx.com/revolution/2.x/getting-started/server-requirement).

I will do the suggested changes later today. 👍

JoshuaLuckers committed:
Prevent users from being blocked to install MODX via composer if ImageMagick is not installed

"Imagick can be replaced (and it does) by php-gd. It works better but it is not installed everywhere so we can not lock MODX to use this extension."
Therefore "ext-gd" is moved to the suggested section.

alroniks commented:
But php-gd is really required for thumbnails and imagick is missed in recommended section.

JoshuaLuckers commented:
> But php-gd is really required for thumbnails and imagick is missed in recommended section.

My misunderstanding!

JoshuaLuckers committed:
Add back ext-gd as required and ext-imagick as suggested extension

alroniks commented:
Not sure, but maybe it does not make sense to install imagic in travis.

JoshuaLuckers committed:
Remove the installation of ImageMagick on Travis

alroniks commented:
Now it looks better. @opengeek

Mark-H reviewed (approved):
Not sure if this can be merged with a `changes requested` review from the other code owner.. but LGTM.

alroniks commented:
@Mark-H We have to wait for approval from @opengeek in this case.

JoshuaLuckers commented:
@opengeek I made the suggested changes.
